File: fe2-android/app/src/test/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/InputTest.java
package com.github.dedis.popstellar.model.network.method.message.data.digitalcash;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;

import com.github.dedis.popstellar.model.objects.security.PublicKey;
import com.github.dedis.popstellar.model.objects.security.Signature;

import org.junit.Test;

public class InputTest {
  private static final int TX_OUT_INDEX = 0;
  private static final String Tx_OUT_HASH = "47DEQpj8HBSa--TImW-5JCeuQeRkm5NMpJWZG3hSuFU=";

  private static final String TYPE = "P2PKH";
  private static final String PUBKEY = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";
  private static final String SIG = "CAFEBABE";

  private static final ScriptInput SCRIPTTXIN = new ScriptInput(TYPE, new PublicKey(PUBKEY), new Signature(SIG));

  private static final Input TXIN = new Input(Tx_OUT_HASH, TX_OUT_INDEX, SCRIPTTXIN);

  @Test
  public void testGetTxOutHash() {
    assertEquals(Tx_OUT_HASH, TXIN.getTxOutHash());
  }

  @Test
  public void testGetTxOutIndex() {
    assertEquals(TX_OUT_INDEX, TXIN.getTxOutIndex());
  }

  @Test
  public void testGetScript() {
    assertEquals(SCRIPTTXIN, TXIN.getScript());
  }

  @Test
  public void testTestEquals() {
    assertEquals(TXIN, new Input(Tx_OUT_HASH, TX_OUT_INDEX, SCRIPTTXIN));
    String random = "random";
    assertNotEquals(TXIN, new Input(random, TX_OUT_INDEX, SCRIPTTXIN));
    assertNotEquals(TXIN, new Input(Tx_OUT_HASH, 4, SCRIPTTXIN));
    assertNotEquals(
        TXIN, new Input(Tx_OUT_HASH, TX_OUT_INDEX, new ScriptInput(random, new PublicKey(PUBKEY), new Signature(SIG))));
  }
}

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/InputTest.kt
package com.github.dedis.popstellar.model.network.method.message.data.digitalcash

import com.github.dedis.popstellar.model.objects.security.PublicKey
import com.github.dedis.popstellar.model.objects.security.Signature
import org.junit.Assert
import org.junit.Test

class InputTest {
  @Test
  fun testGetTxOutHash() {
    Assert.assertEquals(Tx_OUT_HASH, TXIN.txOutHash)
  }

  @Test
  fun testGetTxOutIndex() {
    Assert.assertEquals(TX_OUT_INDEX.toLong(), TXIN.txOutIndex.toLong())
  }

  @Test
  fun testGetScript() {
    Assert.assertEquals(SCRIPTTXIN, TXIN.script)
  }

  @Test
  fun testTestEquals() {
    Assert.assertEquals(TXIN, Input(Tx_OUT_HASH, TX_OUT_INDEX, SCRIPTTXIN))

    val random = "random"

    Assert.assertNotEquals(TXIN, Input(random, TX_OUT_INDEX, SCRIPTTXIN))
    Assert.assertNotEquals(TXIN, Input(Tx_OUT_HASH, 4, SCRIPTTXIN))
    Assert.assertNotEquals(
      TXIN,
      Input(Tx_OUT_HASH, TX_OUT_INDEX, ScriptInput(random, PublicKey(PUBKEY), Signature(SIG)))
    )
  }

  companion object {
    private const val TX_OUT_INDEX = 0
    private const val Tx_OUT_HASH = "47DEQpj8HBSa--TImW-5JCeuQeRkm5NMpJWZG3hSuFU="
    private const val TYPE = "P2PKH"
    private const val PUBKEY = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="
    private const val SIG = "CAFEBABE"
    private val SCRIPTTXIN = ScriptInput(TYPE, PublicKey(PUBKEY), Signature(SIG))
    private val TXIN = Input(Tx_OUT_HASH, TX_OUT_INDEX, SCRIPTTXIN)
  }
}
